docs/tutorials/build-a-dapp/pallet.md

```diff
@@ -104,40 +115,53 @@ use frame_system::ensure_signed;
 use sp_std::vec::Vec;
 ```
 
+> Critical: the `no_std` feature is **required** for all pallets! This is because we are building a
+> _runtime_ module that _must_ compile to WASM, and therefore cannot depend on rust's `std` 
+> dependencies.
+
 Most of these imports are already available because they were used in the template pallet whose code
 we just deleted. However, `sp_std` is not available and we need to list it as a dependency.
 
-**Add** this block to your `pallets/template/Cargo.toml` file under `[dependencies]` section.
+**Add** these lines to your `pallets/template/Cargo.toml` file under `[dependencies]` and 
+`[features]` sections, as shown below:
+
+**`pallets/template/Cargo.toml`**
 
 ```toml
 [dependencies]
+
 #--snip--
-sp-std = { default-features = false, version = '3.0.0' }
-```
 
-Then, **Update** the existing `[features]` block to look like this. The last line is new. You will
-learn more about why this is necessary in the next tutorial, the [Add a Pallet](../add-a-pallet)
-tutorial.
+sp-std = { default-features = false, version = '3.0.0' }  <-- Add this line
+
+#--snip--
 
-```toml
 [features]
 default = ['std']
 std = [
-    'codec/std',
-    'frame-support/std',
-    'frame-system/std',
-    'sp-std/std',          <-- This line is new
+    #--snip--
+    'sp-std/std',                                         <-- Add this line
 ]
 ```
 
-### Configuration
+> Notice here that all _runtime specific_ dependencies _must_ have default feature (meaning `std`)
+> disabled so we can compile the runtime to WASM.You can also learn more about why this is 
+> necessary in the [Add a Pallet](../add-a-pallet) tutorial.
+>
+> You _can_ use `std` features in non-runtime components like `mock.rs` and `tests.rs` using 
```

Comment:
I like the references here. Nice. Just noticed a missing space between sentences :)

Reply:
Also - this could be something to reference additionally: https://substrate.dev/docs/en/knowledgebase/advanced/executor#wasm-execution

Reply:
fix it?